### DISCUSSION TITLE
Ability to change ID column type for a project

### DISCUSSION THREAD

**Discussion by nathanclayton** [nocodb/nocodb#232] [General]
Would it be possible to change the default ID column type for a project?

For example, when using PostgreSQL, changing the default ID column type to `uuid` or `bigserial`?

**Comment by o1lab** [comment]
This is possible but not straight forward. I believe we need to give a setting which would allow one to create from form. 

If you want to do it now : You can go to fields -> show system fields -> checkbox select for ID -> in spreadsheet change the column type.

Thanks for putting here in discussion. Appreciate it.

**Reply by nathanclayton** [reply]
Thank you - I tried that, but depending on the DB, you can't change the PK type after, especially from something like `serial` to `uuid`. I submitted a bug regarding dropping columns of composite primary keys in PG (#234 ).

Would you like me to make additional threads for other feature requests (the largest would be `n:n` relations, it seems like it currently supports `1:n` relations, however that doesn't work if you wanted to support something like a lookup of a limited number of tags, etc.). This would be great to use to dynamically add to/populate multselects throughout the system.

**Reply by o1lab** [reply]
Additional threads would be better to discuss.

We are looking into n:n - should be possible.

**Comment by wingkwong** [comment]
#346 is also related to this thread
